hnn_core/pyramidal.py

```diff
@@ -430,11 +465,11 @@ class L5Pyr(Pyr):
         List of dendrites.
     """
 
-    def __init__(self, gid=-1, pos=-1, p={}):
```

Comment:
defining `p={}` is an anti-pattern. You can get surprising results. See [here](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html)